packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -372,6 +373,39 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange>
 		this.editManager.advanceMinimumSequenceNumber(brand(message.minimumSequenceNumber));
 	}
 
+	/**
+	 * Process a bunch of messages. Some important invariants for these messages:
+	 * - They are all from the same client.
+	 * - They are all part of the same batch so they are not interleaved with messages from other clients.
+	 *-  They are not interleaved with messages from other DDSes in the container.
+	 */
+	protected override processMessagesCore(messagesCollection: IRuntimeMessageCollection): void {
+		const { envelope, local, messagesContent } = messagesCollection;
+		const commits: Commit<TChange>[] = [];
+
+		// Get a list of all the commits from the messages.
+		for (const messageContent of messagesContent) {
+			// Empty context object is passed in, as our decode function is schema-agnostic.
+			const { commit, sessionId } = this.messageCodec.decode(messageContent.contents, {
+				idCompressor: this.idCompressor,
+			});
+			commits.push({ ...commit, sessionId });
+		}
+
+		this.editManager.addSequencedChanges(
+			commits,
+			brand(envelope.sequenceNumber),
+			brand(envelope.referenceSequenceNumber),
+		);
+
+		// Update the resubmit machine for each commit applied.
+		for (const _ of messagesContent) {
+			this.resubmitMachine.onSequencedCommitApplied(local);
```

Comment:
It would be good to update the doc comment for `onSequencedCommitApplied` on the `ResubmitMachine` interface to call out that it may be called multiples times in a row after a number of sequenced commits have been applied (as opposed to always being called before the next sequenced commit is applied).

That has always been true, but we didn't use to rely on that fact before, so it's important that it be document so implementers of the interface are aware.